### PR TITLE
Improve contributing guide

### PR DESCRIPTION
**Pull request by rohitsardesai83**
**What type of PR is this?**
 /kind documentation

**What this PR does / why we need it**:
Improve contributor guide to help community understand the contribution process.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,22 +1,79 @@
...
+
+# Your First Contribution
+
+We will help you to contribute in different areas like filing issues, developing features, fixing critical bugs and getting your work reviewed and merged.
+
+If you have questions about the development process, feel free to jump into our [Slack Channel](https://join.slack.com/t/kubeedge/shared_invite/enQtNjc0MTg2NTg2MTk0LWJmOTBmOGRkZWNhMTVkNGU1ZjkwNDY4MTY4YTAwNDAyMjRkMjdlMjIzYmMxODY1NGZjYzc4MWM5YmIxZjU1ZDI) or join our [mailing list](https://groups.google.com/forum/#!forum/kubeedge).
+
+## Find something to work on
+
+We are always in need of help , be it fixing documentation , reporting bugs or writing some code.
```
Comment (sids-b) [inline CONTRIBUTING.md]: Please remove blank space before ","

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,22 +1,79 @@
...
 - Fork the repository on GitHub
-- Read the README.md for build instructions
+- Read the [setup](https://github.com/kubeedge/kubeedge/blob/master/docs/setup/setup.md) for build instructions
+
+
+# Your First Contribution
+
+We will help you to contribute in different areas like filing issues, developing features, fixing critical bugs and getting your work reviewed and merged.
+
+If you have questions about the development process, feel free to jump into our [Slack Channel](https://join.slack.com/t/kubeedge/shared_invite/enQtNjc0MTg2NTg2MTk0LWJmOTBmOGRkZWNhMTVkNGU1ZjkwNDY4MTY4YTAwNDAyMjRkMjdlMjIzYmMxODY1NGZjYzc4MWM5YmIxZjU1ZDI) or join our [mailing list](https://groups.google.com/forum/#!forum/kubeedge).
```
Comment (sids-b) [inline CONTRIBUTING.md]: I hope the slack invite link is the updated one from the README :) The previous one had expired

Comment (rohitsardesai83) [reply CONTRIBUTING.md]: Its the updated one.

File: CONTRIBUTING.md (hunk 1 above)
Comment (rohitsardesai83) [reply CONTRIBUTING.md]: updated.

**Comment by rohitsardesai83** [comment]
@subpathdev , @sids-b , I have updated as per review comments. PTAL.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,45 +1,113 @@
...
 
-## Getting started
+## Code of Conduct
+
+Please make sure to read and observe our [Code of Conduct](https://github.com/kubeedge/kubeedge/blob/master/CODE_OF_CONDUCT.md).
+
+## Community Expectations
+
+KubeEdge is a community project driven by its community which strives to promote a healthy, friendly and productive environment.
+The goal of the community is to develop a cloud native edge computing platform which takes advantages of the features provided by a popular orchestration technology like Kubernetes to manage edge nodes and devices at scale and demonstrate resiliency, reliability in offline scenarios. To build a platform at such scale requires the support of a community with similar aspirations.
```
Comment (fisherxu) [inline CONTRIBUTING.md]: How about `develop a cloud native edge computing platform built upon kubernetes which takes...`, not like Kubernetes

Comment (rohitsardesai83) [reply CONTRIBUTING.md]: fixed. thanks.

**Comment by rohitsardesai83** [comment]
@fisherxu , @subpathdev , @sids-b , PTAL

**Comment by fisherxu** [comment]
lgtm

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -60,3 +128,16 @@ The format can be described more formally as follows:
 ```
 
 The first line is the subject and should be no longer than 70 characters, the second line is always blank, and other lines should be wrapped at 80 characters. This allows the message to be easier to read on GitHub as well as in various git tools.
+
+Note: if your pull request isn't getting enough attention, you can use the reach out on Slack to get help finding reviewers.
+
```
Comment (chendave) [inline CONTRIBUTING.md]: I am not  a native speaker, so just a soft suggestion, is it good to reword this sentence to something like this, "you can reach out to Slack (link here...) to ask for more attention from reviewers."?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,45 +1,113 @@
...
+# Getting started
 
 - Fork the repository on GitHub
-- Read the README.md for build instructions
+- Read the [setup](https://github.com/kubeedge/kubeedge/blob/master/docs/setup/setup.md) for build instructions.
+
+
+# Your First Contribution
+
```
Comment (chendave) [inline CONTRIBUTING.md]: I am wondering whether the project meeting should be mentioned somewhere in the doc.

File: CONTRIBUTING.md (hunk 4 above)
Comment (rohitsardesai83) [reply CONTRIBUTING.md]: thanks for the feedback @chendave , the slack link has been provided on line 43 above.
I guess "reach out on Slack" sounds correct :)

File: CONTRIBUTING.md (hunk 5 above)
Comment (rohitsardesai83) [reply CONTRIBUTING.md]: I am not sure , it is there in the main README : https://github.com/kubeedge/kubeedge/#meeting
Do we need to specify it here again ?

Comment (chendave) [reply CONTRIBUTING.md]: That's okay, I am just asking. :)

**Pull request merged**